imagebuildah/build.go

```diff
@@ -1663,6 +1666,7 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) (image
 			if !b.layers {
 				cleanupImages = append(cleanupImages, imageID)
 			}
+			imageID = ""
```

Comment:
Is this going to cause us to remember only the first intermediate stage that we run into?  I don't see imageID getting set within this `for stageIndex, stage := range stages ` loop and I'm not sure that's correct?

Reply:
Not sure I follow - `imageID` is set to the value that `stageExecutor.Execute()` returns, so clearing it here is mainly about making sure that if we return early due to an error result from `ib.From()` in a subsequent loop iteration, we don't mistakenly leave the most recent stage in place when, because it's a failure case, we want to remove it.

Reply:
Ugh, false alarm, sorry.  I somehow didn't see imageID being set at line 1645.  I could have sworn I saw it getting cleared here and not reset again within the for loop getting the next stage.  Can I chalk it up to a lack of tea?